st2actions/st2actions/query/mistral/v2.py

```diff
@@ -101,6 +101,11 @@ def _get_workflow_tasks(self, exec_id):
         resp = requests.get(url)
         result = resp.json()
         tasks = result.get('tasks', [])
+
+        for task in tasks:
+            for attr in ['result', 'input', 'output']:
+                task[attr] = jsonify.try_loads(task.get(attr, None))
```

Comment:
Maybe.  It's restricted by the wsme-pecan.  wsme won't allow arbitrary dict and so the API stores and returns them in text.  Also, the query module is the right place to process data if it's not in the format we needed.  It's not always the case we have control over how other API are written.